provider/k8s/k8s.go

```diff
@@ -263,8 +274,28 @@ func (p *Provider) initializeTemplates() error {
 	}
 
 	if p.CertManager {
-		if err := p.applySystemTemplate("cert-manager", nil); err != nil {
-			return errors.WithStack(err)
+		d, _ := p.Cluster.AppsV1().Deployments("cert-manager").Get("cert-manager", am.GetOptions{})
+		if d == nil {
+			return p.applySystemTemplate("cert-manager", nil)
+		}
+
+		if d.Spec.Template.Labels["app.kubernetes.io/version"] != "v1.9.0" {
+			fmt.Println("Updating cert-manager")
+			p.deleteSystemTemplate("cert-manager", nil)
+
+			for {
+				n, _ := p.Cluster.CoreV1().Namespaces().Get("cert-manager", am.GetOptions{})
+				if n.Name == "" {
```

Comment:
should we also check `if n != nil` since it's a pointer?
If `Namespaces().Get()` always return a non nil `n` it's fine though.
Also found this [check](https://github.com/convox/convox/blob/master/provider/k8s/app.go#L97) that it's a bit more idiomatic I think.

Reply:
We should also add a max retry and a backoff time.
something like
```go
maxRetries := 10
retries := 0 // increment by one on each loop iteration
for {
  ...
 time.Sleep(time.Seconds * 10)
}
```

If it doesn't delete the ns within the threshold just print the error and return

Reply:
> should we also check `if n != nil` since it's a pointer? If `Namespaces().Get()` always return a non nil `n` it's fine though. Also found this [check](https://github.com/convox/convox/blob/master/provider/k8s/app.go#L97) that it's a bit more idiomatic I think.

the `Namespaces().Get()` returns an namespace object with an empty metadata, using the `n != nil` doesn't work...

Reply:
Also, for all errors inside `func (p *Provider) initializeTemplates() error ` let's just `panic` them so we still keep the same behavior and fail the API initialization.